src/classes/UTIL_RecordTypes_TEST.cls

```diff
@@ -28,54 +28,72 @@
         
         // check against queried rectypes
-        list<recordtype> ort = [select id, name from recordtype where isactive=true and sobjecttype='Opportunity' limit 1];
+        List<RecordType> ort = [SELECT Id, Name FROM RecordType WHERE IsActive=true AND sObjectType='Opportunity' LIMIT 1];
         if (!ort.isEmpty()) {
             Id rtId = UTIL_RecordTypes.GetRecordTypeId(Opportunity.sObjectType, ort[0].name);
-            system.assertEquals(ort[0].id, rtId); 
+            System.assertEquals(ort[0].id, rtId);
             String rtName = UTIL_RecordTypes_API.GetRecordTypeName('Opportunity', ort[0].id);
-            system.assertEquals(ort[0].name, rtName); 
+            System.assertEquals(ort[0].name, rtName);
 
             // the sets above should contain something too
-            system.assert(oppRtSet.size() > 0);
-            system.assert(rtOptions.size() > 0);            
+            System.assert(oppRtSet.size() > 0);
+            System.assert(rtOptions.size() > 0);
         }       
     }
-    
-    static testMethod void TestRecordTypes() {
+
+    @isTest
+    static void testVFRecordTypes() {
 
         ID idGift = UTIL_RecordTypes.GetRecordTypeId(Opportunity.sObjectType, UTIL_RecordTypes.getRecordTypeNameForGiftsTests(Opportunity.sObjectType));
         
         Set<String> setStr = new Set<String>{UTIL_RecordTypes.getRecordTypeNameForGiftsTests(Opportunity.sObjectType)};
         Set<Id> setId = UTIL_RecordTypes.GetRecordTypeIdSet(Opportunity.sObjectType, setStr);
-        if(idGift!=null){
-            system.Assert(setId.contains(idGift));
+
+        if (idGift!=null) {
+            System.assert(setId.contains(idGift));
         
-            system.AssertEquals(UTIL_RecordTypes.getRecordTypeNameForGiftsTests(Opportunity.sObjectType), UTIL_RecordTypes.GetRecordTypeName(Opportunity.sObjectType, idGift));        
+            System.assertEquals(UTIL_RecordTypes.getRecordTypeNameForGiftsTests(Opportunity.sObjectType), UTIL_RecordTypes.GetRecordTypeName(Opportunity.sObjectType, idGift));
                 
             List<SelectOption> listSO = UTIL_RecordTypes.getRecordTypesForSelectList(Opportunity.sObjectType);
-            system.Assert(listSO.IsEmpty() == false);       
+            System.assert(listSO.IsEmpty() == false);
+        }
+    }
+
+    @isTest
+    static void testLightningRecordTypes() {
+
+        List<RecordType> recordTypes = [SELECT Id, Name FROM RecordType WHERE IsActive=true AND sObjectType='Opportunity' LIMIT 1];
+
+        List<Map<String, String>> recordTypeIdAndName = UTIL_RecordTypes.getRecordTypeDevNamesForLightningSelectList(Opportunity.sObjectType);
+        List<Map<String, String>> recordTypeDevNameAndName = UTIL_RecordTypes.getRecordTypesForLightningSelect(Opportunity.sObjectType);
+        List<Map<String, String>> recordTypeNameAndName = UTIL_RecordTypes.getStringRecordTypesForLightningSelect(Opportunity.sObjectType);
+
+        if (recordTypes.size() > 0) {
```

Comment:
@lmeerkatz I was having some trouble getting the assigned record types here, so instead made a similar test to verify that if there are active record types, then these new lightning select methods should return them. However, if there are no records assigned to the running user, this test will fail. Do you have any suggestions on improving the logic, maybe on a pattern for creating a permission set to make sure that we can assign record types before running these asserts? Or is this ok for now?

Reply:
Are there any record types that are shipped as part of the package? Otherwise, we'll have to make the tests conditional. I believe you can run as a System Administrator user -  `UTIL_UnitTestData_TEST.createUserWithoutInsert('System Administrator')` - and they will have all record types assigned by default (but you'd need to confirm that) but that will only work if multiple record types exist.